### PR TITLE
fix(server): Filter dynamice path(PUT/GET/DELETE) with params cause OOM

### LINKED ISSUE #2566: [Bug] AccessLogFilter for path matrics make Server OOM
### Bug Type (问题类型)

logic (逻辑设计问题)

### Before submit

- [X] 我已经确认现有的 [Issues](https://github.com/apache/hugegraph/issues) 与 [FAQ](https://hugegraph.apache.org/docs/guides/faq/) 中没有相同 / 重复问题 (I have confirmed and searched that there are no similar problems in the historical issue and documents)

### Environment (环境信息)

- Server Version: 1.3.0 (Apache Release Version)
- Backend: RocksDB  

### Expected & Actual behavior (期望与实际表现)

API
```
PUT http://127.0.0.1:8080/graphs/hugegraph/graph/vertices/"1:mako"?action=append

{
    "label": "person",
    "properties": {
        "age": 30,
        "city": "Beijing"
    }
}

```

Get metrics
```
curl http:/127.0.0.1:8080/metrics |grep vertices

result
_graphs_hugegraph_graph_vertices_%221:mako%22_PUT_RESPONSE_TIME_HISTOGRAM
```

when there is many Vetices update

This Map will be oom
ServerReporter.java
```
    private SortedMap<String, Histogram> histograms;
```

### PR DESCRIPTION
**Pull request by JackyYangPassion**
## Purpose of the PR

- fix #2566  

## Verifying these changes

- [ ] Trivial rework / code cleanup without any test coverage. (No Need)
- [ ] Already covered by existing tests, such as *(please modify tests here)*.
- [ ] Need tests and can be verified as follows:
    - xxx

## Does this PR potentially affect the following parts?

- [ ]  Dependencies ([add/update license](https://hugegraph.apache.org/docs/contribution-guidelines/contribute/#321-check-licenses) info & [regenerate_dependencies](../hugegraph-server/hugegraph-dist/scripts/dependency/regenerate_known_dependencies.sh)) 
- [ ]  Modify configurations
- [ ]  The public API
- [ ]  Other affects (typed here)
- [ ]  Nope

## Documentation Status

- [ ]  `Doc - TODO` 
- [ ]  `Doc - Done` 
- [ ]  `Doc - No Need`

### CODE REVIEW

**Comment by imbajin** [comment]
@SunnyBoy-WYH could u take a look for it~

**Comment by SunnyBoy-WYH** [comment]
can we merge the ` requestContext.getUriInfo()`?

> URI uri = requestContext.getUriInfo().getRequestUri();
String path = uri.getRawPath();
String method = requestContext.getMethod();
UriInfo uriInfo = requestContext.getUriInfo();
String path = normalizePath(uriInfo.getPath(),method);
String metricsName = join(path, method);

**Comment by SunnyBoy-WYH** [comment]
Let’s not talk about the oom  for now. for HISTOGRAM metric，seems path without param will be better.? @JackyYangPassion @imbajin

**Comment by SunnyBoy-WYH** [comment]
As a consult, Do you have oom system log and ensure the oom due to this map?

**Review by SunnyBoy-WYH — COMMENTED** [review]
can we del ` LOG.debug ` ?

**Comment by imbajin** [comment]
> can we del `LOG.debug` ?

Thanks, adjust it to `trace` & use `{}` format for it, anymore suggestion?

File: hugegraph-server/hugegraph-api/src/main/java/org/apache/hugegraph/api/filter/AccessLogFilter.java [hunk 1]
```diff
@@ -77,6 +79,27 @@ private String join(String path1, String path2) {
...
+        for (Map.Entry<String, java.util.List<String>> entry : pathParameters.entrySet()) {
+            String key = entry.getKey();
+            String value = entry.getValue().get(0);
+            if ("graph".equals(key)) {
+                newPath = newPath.replace(key, value);
+            }
+            newPath = newPath.replace(value, key);
+        }
+
+        LOG.trace("Original Path: {} New Path: {}", requestPath, newPath);
```
Comment (javeme) [inline hugegraph-server/hugegraph-api/src/main/java/org/apache/hugegraph/api/filter/AccessLogFilter.java:99]: expect `"normalize path, original path: '{}', new path: '{}'"`

File: hugegraph-server/hugegraph-api/src/main/java/org/apache/hugegraph/api/filter/AccessLogFilter.java [hunk 2]
```diff
@@ -88,15 +111,23 @@ public void filter(ContainerRequestContext requestContext,
...
+        int status = responseContext.getStatus();
+        if (status != 500 && status != 415) {
+            MetricsUtil.registerCounter(join(metricsName, METRICS_PATH_TOTAL_COUNTER)).inc();
+        }
+
         if (statusOk(responseContext.getStatus())) {
             MetricsUtil.registerCounter(join(metricsName, METRICS_PATH_SUCCESS_COUNTER)).inc();
         } else {
-            MetricsUtil.registerCounter(join(metricsName, METRICS_PATH_FAILED_COUNTER)).inc();
+            //TODO: The return codes for compatibility need to be further detailed.
```
Comment (javeme) [inline hugegraph-server/hugegraph-api/src/main/java/org/apache/hugegraph/api/filter/AccessLogFilter.java:126]: expect a space "// TODO"

File: hugegraph-server/hugegraph-api/src/main/java/org/apache/hugegraph/api/filter/AccessLogFilter.java [hunk 3]
```diff
@@ -105,8 +136,11 @@ public void filter(ContainerRequestContext requestContext,
             long start = (Long) requestTime;
             long executeTime = now - start;
 
-            MetricsUtil.registerHistogram(join(metricsName, METRICS_PATH_RESPONSE_TIME_HISTOGRAM))
-                       .update(executeTime);
+            if (status != 500 && status != 415) {
```
Comment (javeme) [inline hugegraph-server/hugegraph-api/src/main/java/org/apache/hugegraph/api/filter/AccessLogFilter.java:139]: prefer to add a method is isStatusXx() for a clear meaning and expansion

File: hugegraph-server/hugegraph-api/src/main/java/org/apache/hugegraph/api/filter/AccessLogFilter.java [hunk 4]
```diff
@@ -77,6 +79,27 @@ private String join(String path1, String path2) {
         return String.join(DELIMITER, path1, path2);
     }
 
+    private String normalizePath(ContainerRequestContext requestContext) {
```
Comment (javeme) [inline hugegraph-server/hugegraph-api/src/main/java/org/apache/hugegraph/api/filter/AccessLogFilter.java:82]: mark static?

**Review by imbajin — APPROVED** [review]

**Review by VGalaxies — APPROVED** [review]

**Pull request merged**